edb/lib/math.edgeql

```diff
@@ -28,50 +28,50 @@ math::abs(x: std::anyreal) -> std::anyreal
 
 
 CREATE FUNCTION
-math::ceil(x: std::int64) -> std::float64
+math::ceil(x: std::int64) -> std::int64
 {
     SET volatility := 'IMMUTABLE';
-    USING SQL FUNCTION 'ceil';
+    USING SQL FUNCTION 'edgedb.ceil';
 };
 
 
 CREATE FUNCTION
 math::ceil(x: std::float64) -> std::float64
 {
     SET volatility := 'IMMUTABLE';
-    USING SQL FUNCTION 'ceil';
+    USING SQL FUNCTION 'edgedb.ceil';
 };
 
 
 CREATE FUNCTION
-math::ceil(x: std::decimal) -> std::decimal
+math::ceil(x: std::decimal) -> std::bigint
```

Comment:
I'm not sure about this. Why doesn't it return decimal (like the float64 variant returns float64)?

Reply:
float variant returns float, because we have no other choice. float64 has a vastly larger range than int64.  decimal -> bigint does not have this problem, as bigint is able to represent all integer parts of decimal.  `ceil(float) -> int` in Python 3.

Reply:
So you're breaking the symmetry in variants of the `ceil` function just because we can?

I mean in order to get an `anyint` for `ceil(0.1)` you need an explicit cast as its result is `float64`. For `ceil(0.1n)` you suddenly don't need a cast, it's result is `bigint`.  This asymmetry will propagate through arithmetic expressions, i.e. `(ceil(0.1) + 1) IS anyint` is `false`, but `(ceil(0.1n) + 1n) is anyint` is `true`. 

While this isn't a big deal I don't think this kind of asymmetry is good for variants **of the same function**. So I'm -1.

Reply:
OK, changed signatures to retain the type.